src/libexpr/eval.hh

```diff
@@ -282,6 +285,11 @@ public:
         return *new EvalErrorBuilder<T>(*this, args...);
     }
 
+    /**
+     * A cache for evaluation caches, so as to reuse the same root value if possible
+     */
+    std::map<const Hash, ref<eval_cache::EvalCache>> evalCaches;
```

Comment:
Does this cause `libexpr` to depend implicitly on `libcmd`? I.e. is it still possible to link against `libexpr` but not `libcmd`? (Not sure if anybody cares about that, but it might be an issue.)

Reply:
It looks like that type is defined in `src/libexpr/eval-cache.hh`, so I think we are OK?